docs/design/enforceHostNetwork.md

```diff
@@ -0,0 +1,28 @@
+# Enforcing host network design document
+
+By default, the ceph-csi controller plugin operates on the pod network
+but under some circumstances, like provider mode with a dedicated storage network, it is necessary 
```

Comment:
we need to remove `provider mode` as this is specific to ODF.  we can specific where pod network cannot connect to ceph cluster

Reply:
good point. done